core/src/main/java/software/amazon/awssdk/auth/InstanceProfileCredentialsProvider.java

```diff
@@ -17,102 +17,78 @@
 
 import java.io.IOException;
 import java.net.URI;
-import java.net.URISyntaxException;
+import software.amazon.awssdk.AwsSystemSetting;
 import software.amazon.awssdk.SdkClientException;
-import software.amazon.awssdk.annotation.ReviewBeforeRelease;
 import software.amazon.awssdk.internal.CredentialsEndpointProvider;
-import software.amazon.awssdk.internal.EC2CredentialsUtils;
-import software.amazon.awssdk.util.EC2MetadataUtils;
-import software.amazon.awssdk.utils.SdkAutoCloseable;
+import software.amazon.awssdk.internal.HttpCredentialsUtils;
 
 /**
  * Credentials provider implementation that loads credentials from the Amazon EC2 Instance Metadata Service.
  */
-public class InstanceProfileCredentialsProvider implements AwsCredentialsProvider, SdkAutoCloseable {
-    /**
-     * The client to use to fetch the Amazon ECS credentials.
-     */
-    private final EC2CredentialsProvider credentialsFetcher;
+public class InstanceProfileCredentialsProvider extends HttpCredentialsProvider {
 
-    /**
-     * Create an {@link InstanceProfileCredentialsProvider} using the default configuration. See {@link #builder()} for
-     * customizing the configuration.
-     */
-    @ReviewBeforeRelease("This is inconsistent with client builders. Should this be a static create method?")
-    public InstanceProfileCredentialsProvider() {
-        this(builder());
-    }
+    //TODO: make this private
+    static final String SECURITY_CREDENTIALS_RESOURCE = "/latest/meta-data/iam/security-credentials/";
```

Comment:
Can we do this yet?

Reply:
It's used as part of the integration test when setting up the `EC2MetadataServiceMock`, longer term I'd like to completely replace that mock service with wiremock - didn't want to tackle that in this PR.